src/Codec/ArrayCodec.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace MongoDB\Codec;
+
+use MongoDB\Exception\InvalidArgumentException;
+
+use function array_map;
+use function is_array;
+
+/**
+ * Codec to recursively encode/decode values in arrays.
+ *
+ * @template-implements Codec<array, array>
+ */
+class ArrayCodec implements Codec, KnowsCodecLibrary
+{
+    /** @template-use DecodeIfSupported<array, array> */
+    use DecodeIfSupported;
+    /** @template-use EncodeIfSupported<array, array> */
+    use EncodeIfSupported;
+
+    /** @var CodecLibrary|null */
+    private $library = null;
+
+    public function attachLibrary(CodecLibrary $library): void
+    {
+        $this->library = $library;
+    }
+
+    /** @inheritDoc */
+    public function canDecode($value): bool
+    {
+        return is_array($value);
+    }
+
+    /** @inheritDoc */
+    public function canEncode($value): bool
+    {
+        return is_array($value);
+    }
+
+    /** @inheritDoc */
+    public function decode($value): array
+    {
+        if (! $this->canDecode($value)) {
+            throw InvalidArgumentException::invalidType('$value', $value, 'array');
+        }
+
+        return array_map(
+            /**
+             * @param mixed $item
+             * @return mixed
+             */
+            function ($item) {
+                return $this->getLibrary()->decodeIfSupported($item);
+            },
+            $value
+        );
+    }
+
+    /** @inheritDoc */
+    public function encode($value): array
+    {
+        if (! $this->canEncode($value)) {
+            throw InvalidArgumentException::invalidType('$value', $value, 'array');
+        }
+
+        return array_map(
```

Comment:
The `ArrayCodec` and `ObjectCodec` classes were not part of the previous pull request, and their only purpose is to traverse arrays and objects to ensure values they contain are appropriately handled. However, since they are not needed yet I'll remove them from this PR and introduce them later if they prove to be necessary.